retroarch: 1.8.5 -> 1.9.2

thiagokokada opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

ZHF: https://github.com/NixOS/nixpkgs/issues/144627

Our retroarch derivation isn't update for quite some time. This one bumps it and all cores, with the help of a `update.py` script that should make it easier to add/bump cores in the future.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

thiagokokada committed:
libretro: add update.py script

thiagokokada committed:
libretro: 2020-03-06 -> unstable-2021-11-16

thiagokokada committed:
retroarch: 1.8.5 -> 1.9.13.2

thiagokokada committed:
libretro.citra: fix compilation

thiagokokada committed:
retroarch: fix editorconfig error on hashes.json

thiagokokada commented:
Result of `nixpkgs-review pr 146295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>libretro.play</li>
  </ul>
</details>
<details>
  <summary>79 packages built:</summary>
  <ul>
    <li>kodiPackages.libretro-genplus</li>
    <li>kodiPackages.libretro-mgba</li>
    <li>kodiPackages.libretro-snes9x</li>
    <li>libretro.atari800</li>
    <li>libretro.beetle-gba</li>
    <li>libretro.beetle-lynx</li>
    <li>libretro.beetle-ngp</li>
    <li>libretro.beetle-pce-fast</li>
    <li>libretro.beetle-pcfx</li>
    <li>libretro.beetle-psx</li>
    <li>libretro.beetle-psx-hw</li>
    <li>libretro.beetle-saturn</li>
    <li>libretro.beetle-saturn-hw</li>
    <li>libretro.beetle-snes</li>
    <li>libretro.beetle-supergrafx</li>
    <li>libretro.beetle-vb</li>
    <li>libretro.beetle-wswan</li>
    <li>libretro.bluemsx</li>
    <li>libretro.bsnes-mercury</li>
    <li>libretro.citra</li>
    <li>libretro.desmume</li>
    <li>libretro.desmume2015</li>
    <li>libretro.dolphin</li>
    <li>libretro.dosbox</li>
    <li>libretro.eightyone</li>
    <li>libretro.fbalpha2012</li>
    <li>libretro.fbneo</li>
    <li>libretro.fceumm</li>
    <li>libretro.flycast</li>
    <li>libretro.fmsx</li>
    <li>libretro.freeintv</li>
    <li>libretro.gambatte</li>
    <li>libretro.genesis-plus-gx</li>
    <li>libretro.gpsp</li>
    <li>libretro.gw</li>
    <li>libretro.handy</li>
    <li>libretro.hatari</li>
    <li>libretro.mame</li>
    <li>libretro.mame2000</li>
    <li>libretro.mame2003</li>
    <li>libretro.mame2003-plus</li>
    <li>libretro.mame2010</li>
    <li>libretro.mame2015</li>
    <li>libretro.mame2016</li>
    <li>libretro.mesen</li>
    <li>libretro.meteor</li>
    <li>libretro.mgba</li>
    <li>libretro.mupen64plus</li>
    <li>libretro.neocd</li>
    <li>libretro.nestopia</li>
    <li>libretro.np2kai</li>
    <li>libretro.o2em</li>
    <li>libretro.opera</li>
    <li>libretro.parallel-n64</li>
    <li>libretro.pcsx_rearmed</li>
    <li>libretro.picodrive</li>
    <li>libretro.ppsspp</li>
    <li>libretro.prboom</li>
    <li>libretro.prosystem</li>
    <li>libretro.quicknes</li>
    <li>libretro.sameboy</li>
    <li>libretro.scummvm</li>
    <li>libretro.smsplus-gx</li>
    <li>libretro.snes9x</li>
    <li>libretro.snes9x2002</li>
    <li>libretro.snes9x2005</li>
    <li>libretro.snes9x2010</li>
    <li>libretro.stella</li>
    <li>libretro.stella2014</li>
    <li>libretro.tgbdual</li>
    <li>libretro.thepowdertoy</li>
    <li>libretro.tic80</li>
    <li>libretro.vba-m</li>
    <li>libretro.vba-next</li>
    <li>libretro.vecx</li>
    <li>libretro.virtualjaguar</li>
    <li>libretro.yabause</li>
    <li>retroarch</li>
    <li>retroarchBare</li>
  </ul>
</details>

thiagokokada commented:
Tagging as ZHF since this fixes some build issues (`mame` and `play`).

thiagokokada commented:
Need some help here:
- The cores appear on the `retroarch` (after doing a `retroarch.override { cores = with libretro; [...] }` but I can't them to start a game (they don't appear on the available cores).
- `retroarch` itself is missing resource files (so no icons, wrong font, etc).

aanderse commented:
@thiagokokada I recently started working on `retroarch` and was splitting each of the cores out into their own file. Each core can have any number of customizations and I think it makes sense to treat them separately.

I'll post some of my initial work soon for discussion if that sounds good to you.

hugosenari commented on pkgs/misc/emulators/retroarch/cores.nix:
```diff
@@ -539,17 +459,13 @@ in with lib.licenses;
     license = gpl2Plus;
 
-    extraBuildInputs = [ alsa-lib libGLU libGL portaudio python27 xorg.libX11 ];
+    extraBuildInputs = [ alsa-lib libGLU libGL portaudio python3 xorg.libX11 ];
```
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** :medal_military:  :medal_military:  :medal_military:

thiagokokada commented:
> The cores appear on the retroarch (after doing a retroarch.override { cores = with libretro; [...] }; but I can't them to start a game (they don't appear on the available cores).

The strange thing is, it seems that the cores are doing something (if I change the cores, only the file formats recognized by that core shows). But once I select a content it doesn't show the core to play.

thiagokokada commented:
> > The cores appear on the retroarch (after doing a retroarch.override { cores = with libretro; [...] }; but I can't them to start a game (they don't appear on the available cores).
> 
> The strange thing is, it seems that the cores are doing something (if I change the cores, only the file formats recognized by that core shows). But once I select a content it doesn't show the core to play.

The issue seems to be the `retroarch` bump. If I use `retroarch` 1.8.5 the cores itself works.

thiagokokada committed:
retroarch: 1.9.13.2 -> 1.9.2

From retroarch 1.9.3 and above, it stops loading the cores.

This probably can be fixed, but for now at least this brings the
retroarch to a newer (working) version.

thiagokokada commented:
> The issue seems to be the `retroarch` bump. If I use `retroarch` 1.8.5 the cores itself works.

`retroarch` 1.9.2 works, so using it right now. This at least brings us to a release of retroarch from this year.

risicle commented:
Didn't manage a full rebuild due to time constraints, but managed to build a large amount of the `nixpkgs-review` set on nixos aarch64.

thiagokokada commented:
Result of `nixpkgs-review pr 146295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>libretro.play</li>
  </ul>
</details>
<details>
  <summary>79 packages built:</summary>
  <ul>
    <li>kodiPackages.libretro-genplus</li>
    <li>kodiPackages.libretro-mgba</li>
    <li>kodiPackages.libretro-snes9x</li>
    <li>libretro.atari800</li>
    <li>libretro.beetle-gba</li>
    <li>libretro.beetle-lynx</li>
    <li>libretro.beetle-ngp</li>
    <li>libretro.beetle-pce-fast</li>
    <li>libretro.beetle-pcfx</li>
    <li>libretro.beetle-psx</li>
    <li>libretro.beetle-psx-hw</li>
    <li>libretro.beetle-saturn</li>
    <li>libretro.beetle-saturn-hw</li>
    <li>libretro.beetle-snes</li>
    <li>libretro.beetle-supergrafx</li>
    <li>libretro.beetle-vb</li>
    <li>libretro.beetle-wswan</li>
    <li>libretro.bluemsx</li>
    <li>libretro.bsnes-mercury</li>
    <li>libretro.citra</li>
    <li>libretro.desmume</li>
    <li>libretro.desmume2015</li>
    <li>libretro.dolphin</li>
    <li>libretro.dosbox</li>
    <li>libretro.eightyone</li>
    <li>libretro.fbalpha2012</li>
    <li>libretro.fbneo</li>
    <li>libretro.fceumm</li>
    <li>libretro.flycast</li>
    <li>libretro.fmsx</li>
    <li>libretro.freeintv</li>
    <li>libretro.gambatte</li>
    <li>libretro.genesis-plus-gx</li>
    <li>libretro.gpsp</li>
    <li>libretro.gw</li>
    <li>libretro.handy</li>
    <li>libretro.hatari</li>
    <li>libretro.mame</li>
    <li>libretro.mame2000</li>
    <li>libretro.mame2003</li>
    <li>libretro.mame2003-plus</li>
    <li>libretro.mame2010</li>
    <li>libretro.mame2015</li>
    <li>libretro.mame2016</li>
    <li>libretro.mesen</li>
    <li>libretro.meteor</li>
    <li>libretro.mgba</li>
    <li>libretro.mupen64plus</li>
    <li>libretro.neocd</li>
    <li>libretro.nestopia</li>
    <li>libretro.np2kai</li>
    <li>libretro.o2em</li>
    <li>libretro.opera</li>
    <li>libretro.parallel-n64</li>
    <li>libretro.pcsx_rearmed</li>
    <li>libretro.picodrive</li>
    <li>libretro.ppsspp</li>
    <li>libretro.prboom</li>
    <li>libretro.prosystem</li>
    <li>libretro.quicknes</li>
    <li>libretro.sameboy</li>
    <li>libretro.scummvm</li>
    <li>libretro.smsplus-gx</li>
    <li>libretro.snes9x</li>
    <li>libretro.snes9x2002</li>
    <li>libretro.snes9x2005</li>
    <li>libretro.snes9x2010</li>
    <li>libretro.stella</li>
    <li>libretro.stella2014</li>
    <li>libretro.tgbdual</li>
    <li>libretro.thepowdertoy</li>
    <li>libretro.tic80</li>
    <li>libretro.vba-m</li>
    <li>libretro.vba-next</li>
    <li>libretro.vecx</li>
    <li>libretro.virtualjaguar</li>
    <li>libretro.yabause</li>
    <li>retroarch</li>
    <li>retroarchBare</li>
  </ul>
</details>

thiagokokada commented:
Tested with Xbox One Controller (needs to manual configuration, but works afterwards).

Tested emulators:
- bsnes-mercury-accuracy
- fbalpha2012
- fbneo (segfault)
- gambatte
- genesis-plus-gx
- mame2010
- mednafen_gba
- mednafen_pce
- mednafen_psx (needs BIOS, but seems to work fine)
- mednafen_snes
- mednafen_wsan
- mgba
- mupen64-plus-next
- parallei-n64
- pcsx_rearmed
- quicknes
- sms-plus
- snes9x
- snes9x2002
- snes9x2005
- snes9x2010
- vba_next

So I tested many emulators and except for fbneo everything seems to work fine. I think this is ready to be reviewed.

nrdxp commented:
looks nice, but why not the latest release 1.9.13?

thiagokokada commented:
> looks nice, but why not the latest release 1.9.13?

https://github.com/NixOS/nixpkgs/pull/146295#issuecomment-970681235
https://github.com/NixOS/nixpkgs/pull/146295#issuecomment-970796192
https://github.com/NixOS/nixpkgs/pull/146295#issuecomment-970859165

Now, this probably can be fixed by patching the source code, but I am not sure where it needs to be patched. So for now I decided to settle to 1.9.2 release to at least have some improvements.

thiagokokada committed:
libretro: name -> pname+version

thiagokokada committed:
libretro: clean-up

thiagokokada committed:
libretro: update core licenses

thiagokokada committed:
retroarch: update license

thiagokokada committed:
retroarch,libretro: add myself as maintainer

aanderse commented:
> retroarch,libretro: add myself as maintainer

Can we spark up a `libretro` team of maintainers? I'd join that.

thiagokokada commented:
> > retroarch,libretro: add myself as maintainer
> 
> Can we spark up a `libretro` team of maintainers? I'd join that.

Sure. Can you create an issue?

lucasew commented on pkgs/misc/emulators/retroarch/cores.nix:
```diff
@@ -558,77 +437,47 @@ in with lib.licenses;
-    };
     description = "Port of MAME ~2000 to libretro";
-    license = gpl2Plus;
+    license = "MAME";
```
I think it would be better to create the license then reference that variable here

thiagokokada replied:
This is also how other cores not affected by this PR is using it: https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/emulators/retroarch/cores.nix#L808

Also, AFAIK `lib.licenses` only includes open-source licenses. So MAME license is a no-go.

thiagokokada replied:
Correction: `lib.licenses` only includes licenses from SPDX, and MAME is not in SPDX list: https://spdx.org/licenses/

lucasew commented on pkgs/misc/emulators/retroarch/cores.nix:
```diff
@@ -558,77 +437,47 @@ in with lib.licenses;
-    };
     description = "Port of MAME ~2003 to libretro";
-    license = gpl2Plus;
+    license = "MAME";
```
Same license issue

lucasew reviewed (commented):
Nice work there!

ratsclub reviewed (approved):
It worked correctly here! Thank you for your contribution!

thiagokokada commented:
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>libretro.play</li>
  </ul>
</details>
<details>
  <summary>79 packages built:</summary>
  <ul>
    <li>kodiPackages.libretro-genplus</li>
    <li>kodiPackages.libretro-mgba</li>
    <li>kodiPackages.libretro-snes9x</li>
    <li>libretro.atari800</li>
    <li>libretro.beetle-gba</li>
    <li>libretro.beetle-lynx</li>
    <li>libretro.beetle-ngp</li>
    <li>libretro.beetle-pce-fast</li>
    <li>libretro.beetle-pcfx</li>
    <li>libretro.beetle-psx</li>
    <li>libretro.beetle-psx-hw</li>
    <li>libretro.beetle-saturn</li>
    <li>libretro.beetle-saturn-hw</li>
    <li>libretro.beetle-snes</li>
    <li>libretro.beetle-supergrafx</li>
    <li>libretro.beetle-vb</li>
    <li>libretro.beetle-wswan</li>
    <li>libretro.bluemsx</li>
    <li>libretro.bsnes-mercury</li>
    <li>libretro.citra</li>
    <li>libretro.desmume</li>
    <li>libretro.desmume2015</li>
    <li>libretro.dolphin</li>
    <li>libretro.dosbox</li>
    <li>libretro.eightyone</li>
    <li>libretro.fbalpha2012</li>
    <li>libretro.fbneo</li>
    <li>libretro.fceumm</li>
    <li>libretro.flycast</li>
    <li>libretro.fmsx</li>
    <li>libretro.freeintv</li>
    <li>libretro.gambatte</li>
    <li>libretro.genesis-plus-gx</li>
    <li>libretro.gpsp</li>
    <li>libretro.gw</li>
    <li>libretro.handy</li>
    <li>libretro.hatari</li>
    <li>libretro.mame</li>
    <li>libretro.mame2000</li>
    <li>libretro.mame2003</li>
    <li>libretro.mame2003-plus</li>
    <li>libretro.mame2010</li>
    <li>libretro.mame2015</li>
    <li>libretro.mame2016</li>
    <li>libretro.mesen</li>
    <li>libretro.meteor</li>
    <li>libretro.mgba</li>
    <li>libretro.mupen64plus</li>
    <li>libretro.neocd</li>
    <li>libretro.nestopia</li>
    <li>libretro.np2kai</li>
    <li>libretro.o2em</li>
    <li>libretro.opera</li>
    <li>libretro.parallel-n64</li>
    <li>libretro.pcsx_rearmed</li>
    <li>libretro.picodrive</li>
    <li>libretro.ppsspp</li>
    <li>libretro.prboom</li>
    <li>libretro.prosystem</li>
    <li>libretro.quicknes</li>
    <li>libretro.sameboy</li>
    <li>libretro.scummvm</li>
    <li>libretro.smsplus-gx</li>
    <li>libretro.snes9x</li>
    <li>libretro.snes9x2002</li>
    <li>libretro.snes9x2005</li>
    <li>libretro.snes9x2010</li>
    <li>libretro.stella</li>
    <li>libretro.stella2014</li>
    <li>libretro.tgbdual</li>
    <li>libretro.thepowdertoy</li>
    <li>libretro.tic80</li>
    <li>libretro.vba-m</li>
    <li>libretro.vba-next</li>
    <li>libretro.vecx</li>
    <li>libretro.virtualjaguar</li>
    <li>libretro.yabause</li>
    <li>retroarch</li>
    <li>retroarchBare</li>
  </ul>
</details>

thiagokokada committed:
libretro.play: fix build

thiagokokada commented:
Result of `nixpkgs-review pr 146295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>80 packages built:</summary>
  <ul>
    <li>kodiPackages.libretro-genplus</li>
    <li>kodiPackages.libretro-mgba</li>
    <li>kodiPackages.libretro-snes9x</li>
    <li>libretro.atari800</li>
    <li>libretro.beetle-gba</li>
    <li>libretro.beetle-lynx</li>
    <li>libretro.beetle-ngp</li>
    <li>libretro.beetle-pce-fast</li>
    <li>libretro.beetle-pcfx</li>
    <li>libretro.beetle-psx</li>
    <li>libretro.beetle-psx-hw</li>
    <li>libretro.beetle-saturn</li>
    <li>libretro.beetle-saturn-hw</li>
    <li>libretro.beetle-snes</li>
    <li>libretro.beetle-supergrafx</li>
    <li>libretro.beetle-vb</li>
    <li>libretro.beetle-wswan</li>
    <li>libretro.bluemsx</li>
    <li>libretro.bsnes-mercury</li>
    <li>libretro.citra</li>
    <li>libretro.desmume</li>
    <li>libretro.desmume2015</li>
    <li>libretro.dolphin</li>
    <li>libretro.dosbox</li>
    <li>libretro.eightyone</li>
    <li>libretro.fbalpha2012</li>
    <li>libretro.fbneo</li>
    <li>libretro.fceumm</li>
    <li>libretro.flycast</li>
    <li>libretro.fmsx</li>
    <li>libretro.freeintv</li>
    <li>libretro.gambatte</li>
    <li>libretro.genesis-plus-gx</li>
    <li>libretro.gpsp</li>
    <li>libretro.gw</li>
    <li>libretro.handy</li>
    <li>libretro.hatari</li>
    <li>libretro.mame</li>
    <li>libretro.mame2000</li>
    <li>libretro.mame2003</li>
    <li>libretro.mame2003-plus</li>
    <li>libretro.mame2010</li>
    <li>libretro.mame2015</li>
    <li>libretro.mame2016</li>
    <li>libretro.mesen</li>
    <li>libretro.meteor</li>
    <li>libretro.mgba</li>
    <li>libretro.mupen64plus</li>
    <li>libretro.neocd</li>
    <li>libretro.nestopia</li>
    <li>libretro.np2kai</li>
    <li>libretro.o2em</li>
    <li>libretro.opera</li>
    <li>libretro.parallel-n64</li>
    <li>libretro.pcsx_rearmed</li>
    <li>libretro.picodrive</li>
    <li>libretro.play</li>
    <li>libretro.ppsspp</li>
    <li>libretro.prboom</li>
    <li>libretro.prosystem</li>
    <li>libretro.quicknes</li>
    <li>libretro.sameboy</li>
    <li>libretro.scummvm</li>
    <li>libretro.smsplus-gx</li>
    <li>libretro.snes9x</li>
    <li>libretro.snes9x2002</li>
    <li>libretro.snes9x2005</li>
    <li>libretro.snes9x2010</li>
    <li>libretro.stella</li>
    <li>libretro.stella2014</li>
    <li>libretro.tgbdual</li>
    <li>libretro.thepowdertoy</li>
    <li>libretro.tic80</li>
    <li>libretro.vba-m</li>
    <li>libretro.vba-next</li>
    <li>libretro.vecx</li>
    <li>libretro.virtualjaguar</li>
    <li>libretro.yabause</li>
    <li>retroarch</li>
    <li>retroarchBare</li>
  </ul>
</details>

thiagokokada commented:
Got `play` to build. Tried to run a game and it simply locked up my desktop, but maybe someone else will have more luck.

thiagokokada commented:
> Got `play` to build. Tried to run a game and it simply locked up my desktop, but maybe someone else will have more luck.

Should fix this issue: https://github.com/NixOS/nixpkgs/issues/108988

thiagokokada commented:
@aanderse Can I get your review here?

aanderse reviewed (approved):
This is great! Thank you for your hard work on this! We should coordinate what happens after we merge here. Are you ever on the NixOS matrix chat?

thiagokokada commented:
Ok, thanks, merging it.

> This is great! Thank you for your hard work on this! We should coordinate what happens after we merge here. Are you ever on the NixOS matrix chat?

I am, not really active though. But you can send me a DM. Nick is `@k0kada:matrix.org`.
